src/Pages/Auth/Register/index.js

```diff
@@ -1,15 +1,113 @@
-import React from "react"
+import React, { useState } from "react"
+import { useQuery } from "react-query"
 import { Link } from "react-router-dom"
+import Input from "components/Input"
+import Button from "components/Button"
+import { useI18n } from "providers/I18n"
 
 Register.propTypes = {}
 
 function Register() {
+  const { t } = useI18n()
+
+  const [urlUserCode, setUrlUserCode] = useState(window.location.href)
```

Comment:
Since you are not going to be using setUrlUserCode here, it is best to use the useMemo hook

Reply:
what is the difference?

Reply:
I would like some extra info on this because i dont understand why i should use useMemo

Reply:
Nikos we talked over this in great detail during the lesson last Tuesday. Please Google useMemo vs useState for any further information